internal/gitserver/commands.go

```diff
@@ -1201,62 +1201,25 @@ func (c *clientImplementor) CommitsUniqueToBranch(ctx context.Context, repo api.
 	})
 	defer endObservation(1, observation.Args{})
 
-	args := []string{"log", "--pretty=format:%H:%cI"}
-	if maxAge != nil {
-		args = append(args, fmt.Sprintf("--after=%s", *maxAge))
-	}
-	if isDefaultBranch {
-		args = append(args, "HEAD")
-	} else {
-		args = append(args, branchName, "^HEAD")
+	rng := "HEAD"
+	if !isDefaultBranch {
+		rng = fmt.Sprintf("HEAD..%s", branchName)
 	}
 
-	cmd := c.gitCommand(repo, args...)
-	out, err := cmd.CombinedOutput(ctx)
+	commits, err := c.Commits(ctx, repo, CommitsOptions{
+		Range: rng,
+		After: maxAge.Format(time.RFC3339),
+	})
 	if err != nil {
 		return nil, err
 	}
 
-	commits, err := parseCommitsUniqueToBranch(strings.Split(string(out), "\n"))
-	if authz.SubRepoEnabled(c.subRepoPermsChecker) && err == nil {
-		return c.filterCommitsUniqueToBranch(ctx, repo, commits), nil
-	}
-	return commits, err
-}
-
-func (c *clientImplementor) filterCommitsUniqueToBranch(ctx context.Context, repo api.RepoName, commitsMap map[string]time.Time) map[string]time.Time {
-	filtered := make(map[string]time.Time, len(commitsMap))
-	for commitID, timeStamp := range commitsMap {
-		_, err := c.GetCommit(ctx, repo, api.CommitID(commitID))
-		if !errors.HasType(err, &gitdomain.RevisionNotFoundError{}) {
-			filtered[commitID] = timeStamp
-		}
-	}
-	return filtered
-}
-
-func parseCommitsUniqueToBranch(lines []string) (_ map[string]time.Time, err error) {
-	commitDates := make(map[string]time.Time, len(lines))
-	for _, line := range lines {
-		line = strings.TrimSpace(line)
-		if line == "" {
-			continue
-		}
-
-		parts := strings.SplitN(line, ":", 2)
-		if len(parts) != 2 {
-			return nil, errors.Errorf(`unexpected output from git log "%s"`, line)
-		}
-
-		duration, err := time.Parse(time.RFC3339, parts[1])
-		if err != nil {
-			return nil, errors.Errorf(`unexpected output from git log (bad date format) "%s"`, line)
-		}
-
-		commitDates[parts[0]] = duration
+	commitMap := make(map[string]time.Time)
+	for _, commit := range commits {
+		commitMap[string(commit.ID)] = commit.Committer.Date
 	}
 
-	return commitDates, nil
+	return commitMap, err
```

Comment:
```suggestion
	return commitMap, nil
```

This just reads weird, since right above this we do `if err != nil { return nil, err }` , meaning that there can't be an `err` here. Might as well make it explicit that it's `nil` ?